Svc/PrmDb/test/ut/PrmDbImplTester.cpp

```diff
@@ -346,234 +351,160 @@ namespace Svc {
+                    break;
+                case 3:
+                    ASSERT_EVENTS_PrmFileReadError_SIZE(1);
+                    ASSERT_EVENTS_PrmFileReadError(0, PrmReadError::PARAMETER_VALUE_SIZE, 0, sizeof(U32) + 1);
+                    break;
+                default:
+                    FAIL() << "Reached unknown case";
+            }
+        }
+        // Loop through failure statuses
+        for (FwSizeType i = 0; i < 2; i++) {
+            this->m_errorType = FILE_STATUS_ERROR;
+            // Set various file errors
+            switch (i) {
+                case 0:
+                    this->m_status = Os::File::Status::DOESNT_EXIST;
+                    break;
+                case 1:
+                    this->m_status = Os::File::Status::NOT_OPENED;
+                    break;
+                default:
+                    FAIL() << "Reached unknown case";
+            }
+            // Loop through various field reads
+            for (FwSizeType j = 0; j < 4; j++) {
+                clearEvents();
+                this->m_waits = j;
+                this->m_impl.readParamFile();
+                ASSERT_EVENTS_SIZE(1);
+                switch (j) {
+                    case 0:
+                        ASSERT_EVENTS_PrmFileReadError_SIZE(1);
+                        ASSERT_EVENTS_PrmFileReadError(0,PrmReadError::DELIMITER, 0, this->m_status);
+                        break;
+                    case 1:
+                        ASSERT_EVENTS_PrmFileReadError_SIZE(1);
+                        ASSERT_EVENTS_PrmFileReadError(0, PrmReadError::RECORD_SIZE, 0, this->m_status);
+                        break;
+                    case 2:
+                        ASSERT_EVENTS_PrmFileReadError_SIZE(1);
+                        ASSERT_EVENTS_PrmFileReadError(0, PrmReadError::PARAMETER_ID, 0, this->m_status);
+                        break;
+                    case 3:
+                        ASSERT_EVENTS_PrmFileReadError_SIZE(1);
+                        ASSERT_EVENTS_PrmFileReadError(0, PrmReadError::PARAMETER_VALUE, 0, this->m_status);
+                        break;
+                    default:
+                        FAIL() << "Reached unknown case";
+                }
+            }
+        }
+        this->m_errorType = FILE_DATA_ERROR;
+        for (FwSizeType i = 0; i < 2; i++) {
+            clearEvents();
+            this->m_waits = i;
+            this->m_impl.readParamFile();
+            ASSERT_EVENTS_SIZE(1);
+            switch (i) {
+                case 0:
+                    ASSERT_EVENTS_PrmFileReadError_SIZE(1);
+                    ASSERT_EVENTS_PrmFileReadError(0, PrmReadError::DELIMITER_VALUE, 0, 0xA6);
+                    break;
+                case 1:
+                    ASSERT_EVENTS_PrmFileReadError_SIZE(1);
+                    ASSERT_EVENTS_PrmFileReadError(0, PrmReadError::RECORD_SIZE_VALUE, 0, 16777224);
```

Comment:
What is this magic number?

Reply:
I explained this magic number as best I could in a comment....but there is some magic inherited from the original test.

Reply:
OK.